### PR TITLE
implemented state_and_behaviour.rb

### PR DESCRIPTION
**Pull request by HelenMaherovska**

### CODE REVIEW

**Review by oleksandr-kychun — COMMENTED** [review]
Well done) Btw, you can omit explicit raise of ArgumentError. It would be raised anyway.

**Review by oleksandr-kychun — COMMENTED** [review]
It is better to use getter/setter instead of raw instance variables.

**Comment by HelenMaherovska** [comment]
@ghostinshadow done!

**Review by oleksandr-kychun — COMMENTED** [review]
That's better)

File: lib/topics/state_and_behaviour.rb [hunk 1]
```diff
@@ -2,4 +2,27 @@
 
 # exercise state and behaviour
 module StateAndBehaviour
+  # define class Car
+  class Car
+    attr_accessor :year, :color, :model, :current_speed
+    def self.default_car
+      Car.new {}
```
Comment (vbyno) [inline lib/topics/state_and_behaviour.rb]: what is `self` here?

Comment (HelenMaherovska) [reply lib/topics/state_and_behaviour.rb]: it defines class method

File: lib/topics/state_and_behaviour.rb [hunk 2]
```diff
@@ -2,4 +2,27 @@
 
 # exercise state and behaviour
 module StateAndBehaviour
+  # define class Car
+  class Car
+    attr_accessor :year, :color, :model, :current_speed
+    def self.default_car
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: add empty line before each new method

File: lib/topics/state_and_behaviour.rb [hunk 3]
```diff
@@ -2,4 +2,27 @@
...
+      Car.new {}
+    end
+
+    def initialize(car_attributes)
+      raise ArgumentError, 'Expects hash' unless car_attributes.is_a?(Hash)
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb:14]: empty line after raise... logical split

File: lib/topics/state_and_behaviour.rb [hunk 4]
```diff
@@ -2,4 +2,27 @@
...
+      self.year = car_attributes.fetch(:year, 2015)
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: use @year = instead self.year

Comment (HelenMaherovska) [reply lib/topics/state_and_behaviour.rb]: @sveredyuk at one of the comments above there was told to add getter/setter instead of instance variables

Comment (sveredyuk) [reply lib/topics/state_and_behaviour.rb]: Who said this?

Comment (sveredyuk) [reply lib/topics/state_and_behaviour.rb]: https://github.com/bbatsov/ruby-style-guide#attr_family

Comment (HelenMaherovska) [reply lib/topics/state_and_behaviour.rb]: @sveredyuk  ok, I'll fix it)

File: lib/topics/state_and_behaviour.rb (hunk 3 above)
Comment (HelenMaherovska) [reply lib/topics/state_and_behaviour.rb:14]: @sveredyuk I've fixed it)

File: lib/topics/state_and_behaviour.rb [hunk 5]
```diff
@@ -20,7 +20,7 @@ def initialize(car_attributes)
     end
 
     def speed_up(speed)
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: `more_speed` as argument name will be better (imho)

**Pull request closed without merging**